solvers/NBody/mobility.h

```diff
@@ -54,7 +54,14 @@ class NBody: public libmobility::Mobility{
     this->Nbatch = par.Nbatch;
     this->NperBatch = par.NperBatch;
     if(Nbatch<0) Nbatch = 1;
-    if(NperBatch<0) NperBatch = this->numberParticles;
+    if(NperBatch < 0){
+      if(this->initialized){
```

Comment:
If I understood correctly the issue is that it is illegal to call setParameters before initialization. In that case this check should be in the first line at the root level.

Reply:
It would technically still work as long as the correct parameters were given, but I think it makes it more user-friendly and consistent to do it the way you suggest.